packages/useScript/src/useScript.ts

```diff
@@ -33,19 +71,29 @@ export default function useScript(url: string): boolean {
       }
     } else {
       if (script.getAttribute('data-loaded') === 'true') {
-        setReady(true)
+        onReady()
         // Already loaded, so we can return early
-        return () => {}
+        return noop
+      }
+
+      if (script.getAttribute('data-failed') === 'true') {
+        onError()
+        // Already tried loading, so we can return early
+        return noop
```

Comment:
Just return here - Don't need to return a function.

Reply:
Done!

Please do provide more feedback if you deem necessary. ;)